### PR TITLE
Implement form validation

### PR DESCRIPTION
**Pull request by filwas**
This one was pretty hard, to be honest. It took me a very long time before I found the reportValidity() method within the jungles of JS Documentation. ;) But after I found it, I had no problem with customising the submit button, so that it validates the entries first. 

I have a question though. In formMakerFunctions line 74, we have this: 

`blockName.setAttribute('for', hiddenInput.name);`

But the browser is showing an issue with it, as shown on the attached screenshot. No other input has this problem. Could you tell me what's the cause of this?

Screenshot 2023-07-02 at 18 10 23

### CODE REVIEW

File: js/components/wizardButtonWrapper.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+   function submitClick() {
+      let wizardWrapper = buttonWrapper.parentElement;
+      let wizardScreenWrapper = wizardWrapper.parentElement;
+      let form = wizardWrapper.childNodes[1].firstChild;
+
+      let validityCount = 0;
+      form.querySelectorAll('input').forEach((input) => {
+         if (input.reportValidity()) validityCount++;
+      });
+      if (validityCount < 3) return; //love me some validityCount <3<3<3
```
Comment (Sumick) [inline js/components/wizardButtonWrapper.js]: **issue:** You should display error message (as designed in Figma) for specific input for the user - I guess it's just default HTML validation here?

File: js/components/wizardButtonWrapper.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            wizardStep: newWizardStep,
+            characterObject: props.characterObject,
+            formData: null,
+         }),
+      );
+
+      wizardWrapper.remove();
+   }
+
+   function submitClick() {
```
Comment (Sumick) [inline js/components/wizardButtonWrapper.js]: **nitpick:** I was searching long time for this logic - it should be placed together with form logic which should be placed in `booking-details-step`.

**Review by Sumick — CHANGES_REQUESTED** [review]
There is no validation logic as designed in Figma, and there is a problem with maintainability of this code - you mixed business logic with UI logic, and it's very hard to read now. Try to keep business logic in specific domain directory, `booking-details-step` in this example. And try to split it into files, so anyone can understand what is where, and what does what. Example: `booking-details-step/validation.js`, `booking-details-step/form.js`, then use this logic in one place together with UI maker functions

**Comment by filwas** [comment]
I've removed the `wizardButtonWrapper.js ` function, and decided it'd be better to add a button from the level of wizard's step, not the wizard itself. Form and form validation now have their own separate files. I wasn't sure about moving the form-making function into components folder, since it's a highly specific function. Also, could you please mark "implement step back" task as done? It's already been implemented a couple of PRs back, and I'd have to create a new branch and a new pull with nothing just to complete this task.

**Review by Sumick — APPROVED** [review]
Looks good!

**Pull request merged**